### PR TITLE
Implement nearest enemy evaluation function

### LINKED ISSUE #296: Implement nearest_enemy Evaluation Function
### Description of the task

Implement the nearest_enemy evaluation function.

### Outline the task in more detail

We would like to implement the nearest_enemy evaluation function in our new codebase. It is currently defined in the old codebase at `ai/hl/stp/evaluation/team.cpp`. With respect to the interface of this function (ie. what it takes and received), there should be similar types available in the new codebase. That said, it is possible that this function takes a type that is larger then it needs (ex. taking a `World` when it just needs a `Team` and a `ball`). Basically, try to use the smallest type you possible can; this will make it easier to test, review, develop, and work with in the future. 

Note: The function name should be `camelCase` according to our convention. We should also update the `nearest_friendly` function to follow the same convention

### Acceptance criteria

- [ ] Function nearest_enemy is implemented
- [ ] Function nearest_enemy is unit tested

### PR DESCRIPTION
**Pull request by matthewberends**
### Description

Implemented the nearest enemy evaluation function

### Testing Done

Unit tests written

### Resolved Issues

resolves #296 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/ai/hl/stp/evaluation/team.cpp [hunk 1]
```diff
@@ -4,15 +4,26 @@
 std::optional<Robot> Evaluation::nearest_friendly(const Team friendly_team,
```
Comment (garethellis0) [inline src/thunderbots/software/ai/hl/stp/evaluation/team.cpp]: functions should be `lowerFirstCamelCase` instead of `snake_case`.

**Review by garethellis0 — COMMENTED** [review]
Hmm, what do you think about removing the `nearest_friendly` and `nearest_enemy` functions and just having `nearest_robot` instead? I don't see any real need to have two functions that are functionally exactly the same (ie. I could pass `enemy_team` to `nearest_friendly` and get the same result). This would also simplify testing a bit?

**Comment by matthewberends** [comment]
I also thought of having only the nearest robot function but I wasn't sure 100% sure how exactly these evaluation functions are used. If we are able to combine these into one function I definitely think it is worth doing.

**Comment by garethellis0** [comment]
I think that would make sense. @MathewMacDougall - can you confirm this would be fine?

**Comment by MathewMacDougall** [comment]
Yup that's a good idea. @matthewberends go ahead and combine them :+1:

**Review by MathewMacDougall — APPROVED** [review]
Awesome! Nice call combining the functions. LGTM :+1:

**Pull request merged**